### PR TITLE
MGMT-13138: Improve messaging over host connectivity.

### PR DESCRIPTION
**Pull request by paul-maidment**
Currently we get the following message when the IsConnected check is failing during
installation:

 Host failed to install due to timeout while connecting to host

The following error is generated in the following transitions:
1. installing --> if IsConnected failed
2. preparing to install + preparing successfully --> if IsConnected failed
3. installing + installing in progress --> if 
HostNotResponsiveWhileInstallation condition is failing and we are not in rebooting stage or beyond.

This PR improves this by ensuring that we are clear on whether a timeout of the install or a disconnection has occurred. Moreover we make sure that it is clear which phase the error occurred in (installation, preparation etc)

We also make sure that this status is only checked "pre reboot" as after reboot the agent will not be present to provide this status.

## List all the issues related to this PR

- [ ] New Feature 
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/host/statemachine.go [hunk 1]
```diff
@@ -611,20 +611,6 @@ func NewHostStateMachine(sm stateswitch.StateMachine, th TransitionHandler) stat
...
-		},
-		Condition:        stateswitch.Not(If(IsConnected)),
-		DestinationState: stateswitch.State(models.HostStatusError),
-		PostTransition:   th.PostRefreshHost(statusInfoConnectionTimedOut),
-		Documentation: stateswitch.TransitionRuleDoc{
-			Name:        "Move installing host to error when connection times out",
-			Description: "TODO: Document this transition rule.",
-		},
-	})
-
```
Comment (slaviered) [inline internal/host/statemachine.go:627]: Which transition covers this one now?

Comment (paul-maidment) [reply internal/host/statemachine.go:627]: There is basically another transition with the same state in an or condition. So there was no need for this one.

@ Line 629
`
	sm.AddTransitionRule(stateswitch.TransitionRule{
		TransitionType: TransitionTypeRefresh,
		SourceStates: []stateswitch.State{
			stateswitch.State(models.HostStatusInstalling),
			stateswitch.State(models.HostStatusInstallingInProgress),
		},
		Condition:        stateswitch.Not(If(IsConnected)),
		DestinationState: stateswitch.State(models.HostStatusError),
		PostTransition:   th.PostRefreshHost(statusInfoConnectionTimedOutInstalling),
		Documentation: stateswitch.TransitionRuleDoc{
			Name:        "Move installing host to error when connection times out",
			Description: "TODO: Document this transition rule.",
		},
	})`

File: internal/host/transition_test.go [hunk 2]
```diff
@@ -1318,6 +1318,110 @@ var _ = Describe("Refresh Host", func() {
...
+
+		BeforeEach(func() {
+			mockDefaultClusterHostRequirements(mockHwValidator)
+		})
+
+		mockDisconnectedHost := func(hostStatus string, hostStage *models.HostStage) strfmt.DateTime {
+			passedTime := 90 * time.Minute
+			hostCheckInAt := strfmt.DateTime(time.Now().Add(-MaxHostDisconnectionTime -passedTime))
+			host = hostutil.GenerateTestHost(hostId, infraEnvId, clusterId, hostStatus)
+			host.Inventory = hostutil.GenerateMasterInventory()
```
[the remark this answers is not in the document]
Comment (paul-maidment) [reply internal/host/transition_test.go:1331]: I'll be honest, I just cloned this method from elsewhere as I assumed it to be a prerequisite. I can try to tune it a little.

**Pull request merged**